polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -788,6 +815,28 @@ struct State {
 	clock: Box<dyn Clock + Send + Sync>,
 	assignment_criteria: Box<dyn AssignmentCriteria + Send + Sync>,
 	spans: HashMap<Hash, jaeger::PerLeafSpan>,
+	// Per block, candidate records about how long we take until we gather enough
+	// assignments, this is relevant because it gives us a good idea about how many
+	// tranches we trigger and why.
+	time_started_gathering_assignments:
```

Comment:
the naming is a bit confusing
this is not based on time, but block number

Reply:
Is it better now ?